### ISSUE TITLE
StreamResponseEnumerableFromChatbotAsync wrong role?

### ISSUE THREAD

**Issue by SemperFu** [OkGoDoIt/OpenAI-API-dotnet#119]
```
//Calls the API to get a response, which is appended to the current chat's <see cref="Messages"/> as an <see cref="ChatMessageRole.Assistant"/>

_chat.AppendUserInput(prompt);
  await foreach (string responseChunk in _chat.StreamResponseEnumerableFromChatbotAsync())
                {
                    OutputText.Text += responseChunk;
                }

```

Trying to use the following but when looking at the messages in chat I can see that the role for the responses is marked as User.

![image](https://user-images.githubusercontent.com/1786698/232951538-fac4576d-6af5-41c6-9abe-c7e3929152e3.png)

**Comment by Tryanks** [comment]
I have checked the source code [Conversation.cs#L193](https://github.com/OkGoDoIt/OpenAI-API-dotnet/blob/8bc534bc0163e4ca0ddad21a6d58a57910dc6189/OpenAI_API/Chat/Conversation.cs#L193) and there seems to be no problem.

Is it possible that the OpenAI APIf return value is wrong?

**Comment by NorKaiser** [comment]
same problem

**Comment by Tryanks** [comment]
It also looks like there are no errors in the content returned by the OpenAI API :
```
  "choices": [
    {
      "delta": {
        "role": "assistant"
      },
      "finish_reason": null,
      "index": 0
    }
  ]
```
Then I don't know what the problem is.

Maybe the behavior of [Conversation.cs#L193](https://github.com/OkGoDoIt/OpenAI-API-dotnet/blob/8bc534bc0163e4ca0ddad21a6d58a57910dc6189/OpenAI_API/Chat/Conversation.cs#L193) can be hard-coded as Assistant?

@OkGoDoIt Can you take a look at this?

**Comment by Tryanks** [comment]
> by next token response, it will reassigned to user although it's not response in API stream.

Does this mean that the default value of delta.Role is not `null` but `user`?
Does it depend on the API or class `ChatMessage` initialization?

**Comment by sdcb** [comment]
> > by next token response, it will reassigned to user although it's not response in API stream.
> 
> Does this mean that the default value of delta.Role is not `null` but `user`? Does it depend on the API or class `ChatMessage` initialization?

Yes it seems `ChatMessage`'s default role is `user` instead of `null`:
![image](https://user-images.githubusercontent.com/1317141/233322892-7e93d580-4359-4f12-9d31-1289445c5a37.png)
